### PR TITLE
test (salsah): add headless browser testing on Travis

### LINKED ISSUE #143: Include Salsah Tests in Travis-CI
To run the Salsah tests on Travis-CI will require following:
- a running webapi-server (not a problem)
- a running triplestore (not a problem)
- a running sipi instance (doable with docker; the sipi project can generate this image and push it to the docker hub)
- a running salsah "server" (not a problem)
- run all tests in headless mode (I think; this is somewhat tricky and will need a bit of work)

### LINKED ISSUE #588: Salsah tests not passing since last commit

### PR DESCRIPTION
**Pull request by subotic**
### Summary

- update `chromedriver` to 2.33
- adds running of Salsah browser tests in headless mode on Travis
- adds fixes for alleviating flakiness.
- some tests are ignored as they still fail (see #596)

### Issues

- resolves #143 
- resolves #588 
- new #596

### CODE REVIEW

**Comment by subotic** [comment]
resolves #143

File: salsah/src/test/scala/org.knora.salsah/browser/SearchAndEditSpec.scala [hunk 1]
```diff
@@ -654,11 +654,10 @@ class SearchAndEditSpec extends SalsahSpec {
             // get a list of editing fields
             val editFields = page.getEditingFieldsFromMetadataSection(metadataSection)
 
-            val partOfField = editFields(2)
+            val partOfField = editFields(3)
```
Comment (loicjaouen) [inline salsah/src/test/scala/org.knora.salsah/browser/SearchAndEditSpec.scala]: incunabula is very prone to errors as:
- [imagebase.js:99](https://github.com/dhlab-basel/Knora/blob/develop/salsah/src/public/js/imagebase.js#L99) sorts properties by gui-order and
- [incunabula-onto.ttl](https://github.com/dhlab-basel/Knora/blob/develop/webapi/_test_data/ontologies/incunabula-onto.ttl) declares `incunabula:page`'s properties `pageOf` and `description` with the same `guiOrder` value.

**Comment by subotic** [comment]
@benjamingeer I think that we can merge this. When I'm back from my vacation, I will try to resolve #596.

**Comment by subotic** [comment]
the tests are failing, so no merging for now.

**Comment by subotic** [comment]
@benjamingeer I think this is ready know.

(just ignore codeship. I'm trying out another CI/CD service)

**Comment by subotic** [comment]
yes, correct. thanks.

**Pull request merged**